TRAMS-API/Mapping/GetProjectsResponseMapper.cs

```diff
@@ -0,0 +1,73 @@
+﻿using API.Models.Downstream.D365;
+using API.Models.Upstream.Enums;
+using API.Models.Upstream.Response;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace API.Mapping
+{
+    public class GetProjectsResponseMapper : IMapper<GetProjectsD365Model, GetProjectsResponseModel>
+    {
+        private readonly IMapper<AcademyTransfersProjectAcademy,
+                                 Models.Upstream.Response.GetProjectsAcademyResponseModel> _academyMapper;
+
+        public GetProjectsResponseMapper(IMapper<AcademyTransfersProjectAcademy,
+                                                 Models.Upstream.Response.GetProjectsAcademyResponseModel> academyMapper)
+        {
+            _academyMapper = academyMapper;
+        }
+
+        public GetProjectsResponseModel Map(GetProjectsD365Model input)
+        {
+            if (input == null)
+            {
+                return null;
+            }
+
+            return new GetProjectsResponseModel
+            {
+                ProjectId = input.ProjectId,
+                ProjectName = input.ProjectName,
+                ProjectStatus = ExtractProjectStatus(input),
+                ProjectInitiatorFullName = input.ProjectInitiatorFullName,
+                ProjectInitiatorUid = input.ProjectInitiatorUid,
+                ProjectAcademies = ExtractAcademies(input),
+                ProjectTrusts = ExtractTrusts(input)
+            };
+        }
+
+        private static ProjectStatusEnum ExtractProjectStatus(GetProjectsD365Model input)
+        {
+            return input.ProjectStatus != 0
+                   ? MappingDictionaries.ProjecStatusEnumMap
+                                           .Where(v => v.Value == input.ProjectStatus)
+                                           .FirstOrDefault()
+                                           .Key
+                   : 0;
```

Comment:
Question: I believe the valid project status values are 1 and 2; should be therefore be returning null here instead if no match is found or perhaps have a project status of 0 = Unknown in the enum?

Reply:
Enums are non-nullable types. The default value of an uninitialized enumeration, just like other value types, is zero. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/enum